azure/Guidelines.md

```diff
@@ -477,6 +477,210 @@ Overall, this is a very brittle design that leads to a poor developer experience
+This operation is commonly called the "list operation" for the resource, since the primary content of the response
+is an array (list) of the resource type.  Query parameters on the list operation may be provided to control
+which resources are returned and in what order.
+
+Resource collections can often be arbitrarily large, increasing lookup time as well as the size of the responses being sent over the wire. Therefore, it is important that list operations implement pagination.
+
+:white_check_mark: **DO** provide a list operation for each resource type.
+
+:white_check_mark: **DO** structure the response to a list operation as an object with a top-level array property to contain the collection of resources.
+
+:ballot_box_with_check: **YOU SHOULD** use _value_ as the name of the top-level array property -- there are some allowable exceptions to this guidance.
+
+:white_check_mark: **DO** implement pagination of list operation responses unless there is no possibility of a collection exceeding a size appropriate for a single response.
+In particular, include a top-level property in the response named _nextLink_ that contains an opaque URL to return the next page of results when there are additional items in the collection.
+
+:no_entry: **DO NOT** include a _nextLink_ property in the response if there are no additional items in the collection.
+
+:white_check_mark: **DO** clearly document that resources may be skipped or duplicated across pages of a paginated collection unless the operation has made special provisions to prevent this.
+
+Clients MUST be resilient to collection data being either paged or nonpaged for any given request.
+
+Example:
+```json
+{
+  "value":[
+    { "id": "Item 1","price": 99.95,"sizes": null},
+    { … },
+    { … },
+    { "id": "Item 99","price": 59.99,"sizes": null}
+  ],
+  "nextLink": "{opaqueUrl}"
+}
+```
+
+#### Query options
+
+A number of query parameters that may be supported on a list operation to control which resources are returned and in what order. A summary of these query parameters is shown in the following table, with more detail given below.  All these query parameters are optional.
+
+| Parameter name | type | description |
+| -------------- | ---- | ----------- |
+| _filter_       | string | an expression on the resource type that selects the resources to be returned |
+| _orderby_      | array of string | a list of expressions that specify the order of the returned resources |
+| _skip_         | integer | an offset into the collection of the first resource to be returned |
+| _maxpagesize_  | integer | the maximum number of resources to include in a single response |
+
+:white_check_mark: **DO** treat these query parameter names as case-sensitve.
+
+:no_entry: **DO NOT** prefix any of these query parameter names with "$" (the convention in the [OData standard](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_QueryingCollections)).
```

Comment:
Interesting data point: a quick search on azure-rest-api-specs turns up 583 instances of `"name": "$top"` but just a handful of `"name": "top"`.

Reply:
There was a point when our REST Guidelines were "use OData for everything" and it's had a lasting impact.  We've been pushing newer services away from it because it's kind of frustrating for customers when you call something `$whatever` but don't implement any OData semantics.